internal/http/client.go

```diff
@@ -0,0 +1,77 @@
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+type Client struct {
+	client *http.Client
+}
+
+func NewHTTPClient() *Client {
+
+	tlsConfig, err := CreateTLSClientConfig()
+	if err != nil {
+		return nil
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+	}
+
+	return &Client{
+		client: client,
+	}
+}
+
+func CreateTLSClientConfig() (*tls.Config, error) {
+	var certData, keyData, rootCA []byte
+	var err error
+
+	// TODO: get files from conf
+	certData, err = os.ReadFile("/etc/pki/consumer/cert.pem")
+	if err != nil {
+		return nil, fmt.Errorf("cannot read cert-file: %w", err)
+	}
+
+	keyData, err = os.ReadFile("/etc/pki/consumer/key.pem")
+	if err != nil {
+		return nil, fmt.Errorf("cannot read key-file: %w", err)
+	}
+
+	rootCA, err = os.ReadFile("/etc/rhsm/ca/redhat-uep.pem")
```

Comment:
Well, it could be more complicated. There could be other CA certificates (e.g. in case Satellite is used) and it is recommended to read all CA certificates from this directory. On the other hand I'm not sure if `https://cert.console.redhat.com/api/config-manager/v2/profiles/current` API could provided by Satellite server.

Anyway the `rhc` should recover from such situation (Satellite is used) gracefully and it should not print any error in such case.

Reply:
That's right, that URL didn't provide the information by Satellite. But the API exists for Satellite (it's other URL). `rhc` should recover gracefully for the situation but I think that it also need to print the error if it can not reach the endpoint.